dotCMS/src/main/java/com/dotmarketing/util/ConvertToPOJOUtil.java

```diff
@@ -0,0 +1,364 @@
+
+                link.setInternalLinkIdentifier(map.get("internal_link_identifier"));
+
+                link.setLinkType(map.get("link_type"));
+
+                link.setLinkCode(map.get("link_code"));
+
+                ret.add(link);
+            }
+        }
+
+
+        return ret;
+    }
+
+    /**
+     *
+     * @param results
+     * @return
+     */
+    public static List<Identifier> convertDotConnectMapToIdentifier(final List<Map<String,String>> results)
+            throws ParseException {
+        final List<Identifier> ret = new ArrayList<>();
+        if(results == null || results.size()==0){
+            return ret;
+        }
+
+        for (Map<String, String> map : results) {
+            final Identifier i = new Identifier();
+            i.setAssetName(map.get("asset_name"));
+            i.setAssetType(map.get("asset_type"));
+            i.setHostId(map.get("host_inode"));
+            i.setId(map.get("id"));
+            i.setParentPath(map.get("parent_path"));
+
+            if (map.get("syspublish_date") != null && !map.get("syspublish_date").isEmpty()){
+                i.setSysPublishDate(df.parse(map.get("syspublish_date") ));
+            }
+
+            if (map.get("sysexpire_date") != null && !map.get("sysexpire_date").isEmpty()){
+                i.setSysExpireDate(df.parse(map.get("sysexpire_date") ));
+            }
+
+            ret.add(i);
+        }
+        return ret;
+    }
+
+    public static List<Folder> convertDotConnectMapToFolder(final List<Map<String,String>> results)
+            throws ParseException {
+
+        Folder folder;
+        List<Folder> ret;
+
+        ret = new ArrayList<>();
+
+        if(results != null && !results.isEmpty()){
+            for (Map<String, String> map : results) {
+                folder = new Folder();
+                folder.setInode(map.get("inode"));
+                folder.setName(map.get("name"));
+                folder.setTitle(map.get("title"));
+
+                if (map.get("show_on_menu") != null && !map.get("show_on_menu").isEmpty()) {
+                    folder.setShowOnMenu(Boolean.parseBoolean(map.get("show_on_menu")));
+                }
+
+                if (map.get("sort_order") != null && !map.get("sort_order").isEmpty()){
+                    folder.setSortOrder(Integer.parseInt(map.get("sort_order")));
+                }
+
+                folder.setFilesMasks(map.get("files_masks"));
+
+                folder.setIdentifier(map.get("identifier"));
+
+                folder.setDefaultFileType(map.get("default_file_type"));
+
+                if (map.get("mod_date") != null && !map.get("mod_date").isEmpty()){
+                    folder.setModDate(df.parse(map.get("mod_date")));
+                }
+
+                ret.add(folder);
+            }
+        }
+        return ret;
+    }
+
+    public static List<Container> convertDotConnectMapToContainer(final List<Map<String,String>> results)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)